File: java/exercises/01-beginner/ex035_junit_assertions/CalculatorTest.java
package fewolearning.exercises.beginner.ex035_junit_assertions;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CalculatorTest {

    @Test
    void addReturnsTheSumOfTwoNumbers() {
        // Arrange
        int left = 4;
        int right = 5;

        // Act
        int result = Calculator.add(left, right);

        // Assert
        assertEquals(9, result);
    }

    @Test
    void addHandlesNegativeNumbers() {
        assertEquals(-1, Calculator.add(-5, 4));
    }

    @Test
    void subtractReturnsTheDifferenceOfTwoNumbers() {
        // Arrange
        int left = 10;
        int right = 3;

        // Act
        int result = Calculator.subtract(left, right);

        // Assert
        assertEquals(7, result);
    }

    @Test
    void subtractCanProduceANegativeResult() {
        assertEquals(-4, Calculator.subtract(1, 5));
    }
}

File: kotlin/exercises/01-beginner/ex032_junit_assertions/CalculatorTest.kt
package fewolearning.exercises.beginner.ex032_junit_assertions

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class CalculatorTest {

    @Test
    fun addReturnsTheSumOfTwoNumbers() {
        // Arrange
        val left = 2
        val right = 3

        // Act
        val result = add(left, right)

        // Assert
        assertEquals(5, result)
    }

    @Test
    fun subtractReturnsTheDifferenceOfTwoNumbers() {
        // Arrange
        val left = 10
        val right = 4

        // Act
        val result = subtract(left, right)

        // Assert
        assertEquals(6, result)
    }

    @Test
    fun subtractCanReturnANegativeResult() {
        // Arrange
        val left = 3
        val right = 10

        // Act
        val result = subtract(left, right)

        // Assert
        assertEquals(-7, result)
    }
}
